### ISSUE TITLE
Implement Neural Network

### ISSUE THREAD

**Issue by Wojtechnology** [Wojtechnology/ML#6]
Implement a Neural Network Model following the following steps:
- [x] 1) Create interface that randomly initializes weights (should be super flexible)
- [x] 2) Implement forward prop
- [x] 4) Implement backprop to compute partial derivatives
- Perform forward and backprop using example x^i, y^i (Get activations a^l and delta terms d^l for l = 2...L)
- [x] 6) Use gradient descent with back propagation (or find optimized algorithm)
- [ ] 3) Implement code to compute cost function J(theta)
- [ ] 5) Use gradient checking to compare partial derivatives

**Comment by Wojtechnology** [comment]
Just gradient checking left. Opening separate issue.
